tests/data_context/test_data_context_state_management.py

```diff
@@ -30,6 +31,19 @@ def set(self, key, value, **kwargs):
         return super().set(key=key, value=value, **kwargs)
 
 
+class ProfilerStoreSpy(ProfilerStore):
```

Comment:
https://blog.bitsrc.io/unit-testing-deep-dive-what-are-stubs-mocks-spies-and-dummies-6f7fde21f710

Reply:
it's essentially a stub or mock that has the ability to track something specific (method calls in this case).

Probably not the only way to do it but since this is testing "state management", I wanted to track these calls